File: hw/pazmino/u3/HW37StrategyExercise/HW38StrategySortApp/src/main/java/ec/edu/espe/hw38strategysortapp/model/InsertionSort.java
package ec.edu.espe.hw38strategysortapp.model;

/**
 *
 * @author Brandon Pazmino
 */
public class InsertionSort implements SortingStrategy {

    @Override
    public int[] sort(int[] data) {
        System.out.println("Using InsertionSort...");
        for (int i = 1; i < data.length; i++) {
            int key = data[i];
            int j = i - 1;
            while (j >= 0 && data[j] > key) {
                data[j + 1] = data[j];
                j--;
            }
            data[j + 1] = key;
        }
        return data;
    }
}

File: EXAM/pazmino/u3/PazmiñoStrategy/ec/edu/espe/pazmiñostrategy/model/insertion_sort.py
class SortingStrategy:
    def sort(self, data):
        pass

class InsertionSort(SortingStrategy):
    def sort(self, data):
        size = len(data)
        for i in range(1, size):
            key = data[i]
            j = i - 1
            while j >= 0 and data[j] > key:
                data[j + 1] = data[j]
                j -= 1
            data[j + 1] = key
        return data
